### ISSUE TITLE
Learning Schedule for Fast Conformer

### ISSUE THREAD

**Issue by qhoangdl** [NVIDIA/NeMo#8573]
Hi,

I don't see the max_steps specified in this [config](https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/fastconformer/fast-conformer_ctc_bpe.yaml#L184) for Fast-Conformer CTC BPE. Does that mean there's only one cycle in the whole training process because the max_steps param will be the total training steps - warmup_steps?

Do I need to specify the max_steps to something like the total number of iteration per epoch?

Thanks!

**Comment by titu1994** [comment]
Yes, only one cycle. It is preferred to use max steps since it makes it easier to configure the scheduler but you can provide other args and it will work fine

**Comment by qhoangdl** [comment]
Thank you for the clarification!

Regarding model averaging, the paper mentions taking average of the last 5 checkpoint. In your implementation, do you take average of the last 5 checkpoints or simply take average of the 5 best checkpoints?

**Comment by titu1994** [comment]
Last 5 checkpoints is more stable and gets adequate improvements. Best 5 can be unstable based on how "stale" the checkpoints are - potentially an epoch 50 xheckpoint can somehow be merged with epoch 500 and that usually causes degradation in performance

**Comment by qhoangdl** [comment]
I see. And the val_check_interval is 1.0 so the last 5 checkpoints are the model at the end of the last 5 epochs, right?

**Comment by titu1994** [comment]
Usually, but not always. It's usually within the bracket of last 5-15 Epochs

**Comment by qhoangdl** [comment]
I see. Thank you very much!
